homeassistant/components/tis_control/config_flow.py

```diff
@@ -0,0 +1,70 @@
+"""Config flow for TISControl integration."""
+
+from __future__ import annotations
+
+import logging
+
+import voluptuous as vol
+
+from homeassistant.config_entries import ConfigFlow, ConfigFlowResult
+from homeassistant.const import CONF_PORT
+from homeassistant.core import callback
+
+from .const import DOMAIN
+
+_LOGGER = logging.getLogger(__name__)
+
+STEP_USER_DATA_SCHEMA = vol.Schema({vol.Required(CONF_PORT): str}, required=True)
+
+
+class TISConfigFlow(ConfigFlow, domain=DOMAIN):
+    """Handle a config flow for TISControl."""
+
+    VERSION = 1
+
+    async def async_step_user(self, user_input=None) -> ConfigFlowResult:
+        """Handle a flow initiated by the user."""
+        errors = {}
+        if user_input is not None:
+            logging.debug("recieved user input %s", user_input)
+            # Example validation: Check if the port is not empty
+            if not user_input.get(CONF_PORT):
+                errors["base"] = "port_required"  # Custom error key
+                logging.error("Port is required but was not provided.")
+
+            else:
+                # Assuming a function `validate_port` that returns True if the port is valid
+                is_valid = await self.validate_port(user_input[CONF_PORT])
+                if not is_valid:
+                    errors["base"] = "invalid_port"  # Custom error key
```

Comment:
Why do we only have to fill in a port? What do we do with it?

Reply:
because some devices use different port for communication

Reply:
Can you elaborate on that?

Reply:
basically the integration communicates with a gateway and the company manfactures several types of them each one uses a specific port

Reply:
Can we make this more user friendly than have the user fill in a port?

Reply:
unfortunatley the only way to communicate with these devices is by specifying the port manually (can't be auto discovered) so the user must fill it

Reply:
The user can select their device

Reply:
makes sense, but the devices are configurable so that you could change their communication port (can't solve this one)
so if a user had device X that communicates on port Y by default and he changed it to Z then the integration won't work.